File: test/08_test.dart
import 'dart:io';
import 'package:test/test.dart';
import 'package:advent_of_code_2020/08.dart';

void main() {
  String contents = new File('./data/08.txt').readAsStringSync();

  String example = 'nop +0\nacc +1\njmp +4\nacc +3\njmp -3\nacc -99\nacc +1\njmp -4\nacc +6';
  
  group('A', () {
    test('Example 1', () {
      expect(A08(example), equals(5));
    });
    test('Solution', () {
      expect(A08(contents), equals(1941));
    });
  });

  group('B', () {
    test('Example 1', () {
      expect(B08(example), equals(8));
    });
    test('Solution', () {
      expect(B08(contents), equals(2096));
    });
  });
}

File: test/08.test.js
const fs = require('fs');

const { A08, B08 } = require('../lib/08');
const input = fs.readFileSync('./data/08.txt', 'utf-8');

const example = `nop +0
acc +1
jmp +4
acc +3
jmp -3
acc -99
acc +1
jmp -4
acc +6`;

describe('08', () => {
	describe('A', () => {
		it('Example 1', () => {
			expect(A08(example)).toBe(5);
		});

		it('Solution', () => {
			expect(A08(input)).toBe(1941);
		});
	});

	describe('B', () => {
		it('Example 1', () => {
			expect(B08(example)).toBe(8);
		});

		it('Solution', () => {
			expect(B08(input)).toBe(2096);
		});
	});
});
